packages/app/src/components/VisualizationOptions/Options/PositiveNumberBaseType.js

```diff
@@ -0,0 +1,64 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import { InputField } from '@dhis2/ui'
+
+import { sGetUiOptions } from '../../../reducers/ui'
+import { acSetUiOptions } from '../../../actions/ui'
+import { tabSectionOption } from '../styles/VisualizationOptions.style.js'
+
+export const PositiveNumberBaseType = ({
+    label,
+    placeholder,
+    helpText,
+    width,
+    option,
+    value,
+    onChange,
+}) => (
+    <div className={tabSectionOption.className}>
+        <div>
+            <InputField
+                type="number"
+                label={label}
+                onChange={({ value }) => {
+                    const parsedValue = parseInt(value, 10)
+                    parsedValue >= 0
+                        ? onChange(Math.round(value))
+                        : onChange(parsedValue ? 0 : null)
+                }}
```

Comment:
There's a small bug here that could make it return negative numbers if the input is between 0 and -1.

E.g. if you pass "-0.7" then `parsedValue >= 0` will be true, but Math.round will return -1.

If you pass "-0.3" so that the value is floored Math.round will return -0 (which is the same as 0, but not so nice).

Reply:
Suggestion:
`
const parsedValue = parseInt(value, 10)
onChange(isNaN(parsedValue) ? 0 : Math.max(parsedValue, 0))
`

Reply:
Thanks for pointing this out. However when testing (OSX Chrome) the example values above it seems to work as expected and always return `0`. I believe `parsedValue >= 0` prevents anything with `-` and thus reverts to `0` instead. Or am I missing something?